posydon/visualization/plot_pop.py

```diff
@@ -245,9 +246,11 @@ def plot_popsyn_over_grid_slice(pop, grid_type, met_Zsun, slices=None, channel=N
         met_indices = np.where(pop.select(where=channel_sel, columns=['metallicity']) == met_Zsun)[0].tolist()
         if 'HMS-HMS' in grid_path:
             slice_3D_var_range = (dq_edges[i],dq_edges[i+1])
-
-            sel = 'index in '+str(met_indices)+' & event == "ZAMS"'
-            data = pop.history.select(where=sel, columns=['S1_mass','S2_mass' ,'orbital_period'])
+            if len(met_indices) == 0:
+                data = pd.DataFrame(columns=['S1_mass','S2_mass' ,'orbital_period'])
```

Comment:
Wouldn't this be a good place to add a warning, that there is no data to plot? (the same for the other two cases)

Reply:
Additionally, do we still want the plotting or should we skip it (just adding a `continue` instead of creating a new data frame), like `# TODO: skip plotting slice if there are no data` suggests?

Reply:
I find it useful to see the slice even if no binaries are on it, since this helps with verification.

Reply:
OK, please check, whether the other condition `if slices is not None and round(var,2) not in np.around(slices,2)`, should be modified, to produce a plot there, too.